packages/ra-core/src/form/useSuggestions.ts

```diff
@@ -0,0 +1,192 @@
+import { useCallback, isValidElement } from 'react';
+import set from 'lodash/set';
+import useChoices, { UseChoicesOptions } from './useChoices';
+import { useTranslate } from '../i18n';
+
+const defaultMatchSuggestion = getChoiceText => (filter, suggestion) => {
+    const suggestionText = getChoiceText(suggestion);
+
+    const isReactElement = isValidElement(suggestionText);
+
+    return isReactElement
+        ? false
+        : suggestionText.match(
+              // We must escape any RegExp reserved characters to avoid errors
+              // For example, the filter might contains * which must be escaped as \*
+              new RegExp(escapeRegExp(filter), 'i')
+          );
+};
+
+const useSuggestions = ({
+    allowEmpty,
+    choices,
+    emptyText = '',
+    emptyValue = null,
+    limitChoicesToValue,
+    matchSuggestion,
+    optionText,
+    optionValue,
+    selectedItem,
+    suggestionLimit = 0,
+    translateChoice,
+}: Options) => {
+    const translate = useTranslate();
+    const { getChoiceText, getChoiceValue } = useChoices({
+        optionText,
+        optionValue,
+        translateChoice,
+    });
+
+    const getSuggestions = useCallback(
+        filter =>
+            getSuggestionsFactory({
```

Comment:
`getSuggestionsFactory` returns a function that takes `filter` as a parameter, so your `getSuggestions` callback signature will be `filter => filter => choices`? Something doesn't make sense.

Reply:
No because I pass `filter` to the result function of `getSuggestionsFactory`. This is unnecessary however. Left over from debugging sessions ^^